onnxruntime/core/providers/webgpu/shader_variable.cc

```diff
@@ -0,0 +1,327 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <memory>
+#include <string>
+#include <sstream>
+
+#include "core/common/safeint.h"
+#include "core/providers/webgpu/shader_variable.h"
+
+#include "core/providers/webgpu/string_macros.h"
+
+namespace onnxruntime {
+namespace webgpu {
+
+namespace {
+constexpr static const std::string_view STORAGE_TYPE[] = {
```

Comment:
>STORAGE_TYPE [](http://example.com/codeflow?start=40&length=12)

You may consider changing them to std::array. The we will have range check when indexing them.